models/AmazonS3.cfc

```diff
@@ -85,13 +132,73 @@ component accessors="true" singleton {
 		buildUrlEndpoint();
 
 		// Build signature utility
-        variables.sv4Util = new Sv4Util(
-            accessKeyId        = variables.accessKey,
-            secretAccessKey    = variables.secretKey,
-            defaultRegionName  = arguments.awsRegion,
-            defaultServiceName = "s3"
-		);
+		if( variables.signature == "V4" ) {
```

Comment:
I think the approach we'd like to take supporting multiple signatures is to create a signature interface.  To start, I think it would need one method:

```
public struct function generateSignatureData(
        required string requestMethod
        , required string hostName
        , required string requestURI
        , required any requestBody
        , required struct requestHeaders
        , required struct requestParams
        , required string accessKey
        , required string secretKey
        , required string regionName
        , required string serviceName
        , boolean signedPayload = true
        , array excludeHeaders = []
        , string amzDate
        , string dateStamp
    )
```

If your implementation needs different fields, let's see what common fields we can support.

Then here in the constructor we can create the signature object and then use it later without having to do a bunch of `if`/`else` checks.

Does that make sense?

Something here like:

```
variables.signatureUtil = createSignatureUtil( arguments.signatureType );
```

and then used down below as `variables.signatureUtil`.

Reply:
Most of the arguments aren't necessary for v2, but I've made the Sv2Util similar to Sv4Util so it will pass back the same values.